exercises/ciara_nightingale/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,118 @@
+#include <algorithm>
+#include <ctype.h>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <ostream>
+#include <string>
+#include <vector>
+#include <sstream>
+
+// replaces the hypens with whitespace - used in clean_word
+bool replace_hyphens(std::string& word) {
+	bool test = false; //so the whole word is checked rather than stopping at the first instance of a hyphen
+	for (std::string::size_type i = 0; i != word.size(); ++i) {
+		//check through each character in the word
+		if (word[i] == '-') {
+				word[i] = ' ';
+				test = true;
+		}
+  }
+	if (test == true) {
+			return true;
+	}
+  return false;
+}
+
+// determine if a character is punctuation - used in clean_word
+bool is_punc(char const c) {
+  static const std::string punc = ".,?\'\"!();:";
+  for (std::string::size_type i = 0; i != punc.size(); ++i) {
+    if (c == punc[i]) {
+      return true;
+      break;
+    }
+  }
+  return false;
+}
+
+// deal with punctuation and case sensitivity - used in process_word
+void clean_word(std::string& word) {
+  // remove characters if they are puctuation
+  word.erase(remove_if(word.begin(), word.end(), is_punc), word.end());
+  // convert characters to lowercase if they are uppercase
+  transform(word.begin(), word.end(), word.begin(), tolower);
+}
+
+//clean the words, count them and put them into the map
+void processWord(std::string& word, std::map<std::string, int>& counter, std::string::size_type& maxlen, std::string::size_type const min_word_length) {
+		clean_word(word);
+		maxlen = std::max(maxlen, word.size()); //find the longest word.
```

Comment:
To minimise the number of operations we're doing, this line could be moved inside the `if` block below. We don't care about the length of words we're not adding to our counter

Reply:
Correct me if I am wrong but doesn't the clean word function change the length of the string and so moving it inside the if statement would mean that three letter long words with punctuation would accidentally get through.

Reply:
You're right, you definitely want the `clean_word` function outside the `if` statement. I was referring to line 50 where you're finding the max word length.

Reply:
That makes more sense! Thank you.